services/frontend/main.go

```diff
@@ -44,41 +40,48 @@ func run() error {
 		return err
 	}
 
-	usecases := &usecases.Usecases{
-		ContextBox: contextBoxConnector,
-		Done:       make(chan struct{}),
-	}
-
-	k8sSidecarNotificationsReceiver, err := inboundAdapters.NewK8sSidecarNotificationsReceiver(usecases)
-	if err != nil {
-		return err
-	}
-
 	// Start Kubernetes liveness and readiness probe responders
 	healthcheck.NewClientHealthChecker(fmt.Sprint(healthcheckPort),
 		func() error {
-			err := k8sSidecarNotificationsReceiver.PerformHealthCheck()
-			if err != nil {
-				return err
-			}
-
 			return contextBoxConnector.PerformHealthCheck()
 		},
 	).StartProbes()
 
 	errGroup, errGroupContext := errgroup.WithContext(context.Background())
+	usecaseContext, usecaseCancel := context.WithCancel(context.Background())
+
+	usecases := &usecases.Usecases{
+		ContextBox:    contextBoxConnector,
+		SaveChannel:   make(chan *usecases.RawManifest),
+		DeleteChannel: make(chan *usecases.RawManifest),
+		Context:       usecaseContext,
+	}
+
+	secretWatcher, err := inboundAdapters.NewSecretWatcher(usecases)
+	if err != nil {
+		usecaseCancel()
+		return err
+	}
 
-	// Start receiving notifications from the k8s-sidecar container
+	// Start watching for any input manifests and process them as needed.
 	errGroup.Go(func() error {
-		log.Info().Msgf("Listening for notifications from K8s-sidecar at port: %v", k8sSidecarNotificationsReceiverPort)
-		return k8sSidecarNotificationsReceiver.Start("0.0.0.0", k8sSidecarNotificationsReceiverPort)
+		log.Info().Msgf("Frontend is ready to process input manifests")
+		go usecases.ProcessManifestFiles()
+
+		log.Info().Msgf("Frontend is ready to watch input manifest statuses")
+		go usecases.WatchConfigs()
+
+		log.Info().Msgf("Frontend is watching for any new input manifest")
```

Comment:
I think this comment - **Frontend is watching for any Kubernetes secrets with label "claudie.io/input-manifest" in namespace "claudie"** will make more sense.

Reply:
The `main.go` doesn't really have access to the variables like `label` or `namespace`. We can export it in some way, however, I don't see much value in that.